src/main/java/net/mcreator/ui/minecraft/states/block/JBlockPropertiesStatesList.java

```diff
@@ -108,25 +108,81 @@ public JBlockPropertiesStatesList(MCreator mcreator, IHelpContext gui,
 		super.setEnabled(enabled);
 
 		addProperty.setEnabled(enabled);
+		addExisting.setEnabled(enabled);
 
 		propertiesList.forEach(e -> e.setEnabled(enabled));
 
 		propertyEntries.setEnabled(enabled);
 	}
 
+	String propertyRegistryName(PropertyData<?> data) {
+		String name = data.getName();
+		return !name.startsWith("CUSTOM:") ? registryNames.get(name) : name.replace("CUSTOM:", "");
+	}
+
 	private void createPropertiesEntry() {
 		PropertyDataWithValue<?> newEntry = AddBlockPropertyDialog.showDialog(mcreator,
-				propertiesList.stream().map(JBlockPropertiesListEntry::getPropertyData).collect(Collectors.toList()),
-				nonUserProvidedProperties);
+				() -> propertiesList.stream().map(e -> propertyRegistryName(e.getPropertyData()))
+						.filter(Objects::nonNull), nonUserProvidedProperties);
 		if (newEntry != null) {
 			addPropertiesEntry(newEntry);
 		}
 	}
 
+	private void addPropertyFromDataList() {
+		DataListEntry property = DataListSelectorDialog.openSelectorDialog(mcreator,
+				w -> DataListLoader.loadDataList("blockstateproperties"),
+				L10N.t("elementgui.block.custom_properties.add_existing"),
+				L10N.t("elementgui.block.custom_properties.add_existing.message"));
+		if (property == null)
+			return;
+
+		String registryName;
+		if (property.getOther() instanceof Map<?, ?> other && other.get("registry_name") != null)
+			registryName = (String) other.get("registry_name");
+		else
+			return;
+
+		for (JBlockPropertiesListEntry p : propertiesList) {
+			if (registryName.equals(propertyRegistryName(p.getPropertyData()))) {
+				JOptionPane.showMessageDialog(mcreator,
+						L10N.t("elementgui.block.custom_properties.add.error_duplicate"),
+						L10N.t("elementgui.block.custom_properties.add.error_duplicate.title"),
+						JOptionPane.ERROR_MESSAGE);
+				return;
+			}
+		}
+
+		registryNames.put(property.getName(), registryName);

```

Comment:
I would advise to have helper method to convert property name to registry name instead of using this cache that is also currently broken (see my comment above)

Reply:
A helper method like that is already added in this PR, should it instead plainly extract registry name from DL entry with matching name?

Reply:
Adding to this question that it should be accessible from `AddBlockPropertyDialog` and thus this class is not really suitable for this method anymore.

Reply:
> A helper method like that is already added in this PR, should it instead plainly extract registry name from DL entry with matching name?

Yes, I meant that, so instead of cache, you directly extract it from DL entry

> Adding to this question that it should be accessible from AddBlockPropertyDialog and thus this class is not really suitable for this method anymore.

Possibly, but then again I can't find more suitable location.

Reply:
Would `Block` storage class be appropriate location? Alternatively I can add more methods to DLE class itself (like `getOther(String)` and `getOtherOr(String, Object)`), with just this we can shrink the helper method into one line.

Reply:
Block.. no, this is not remplate related. They use mappings.

DLE not so much either (not the right purpose). I would just make a static method here for now.